website/docs/opengl/1-hello-window.md

```diff
@@ -0,0 +1,263 @@
+
+```cs
+_window = Window.Create(options);
+```
+
+Now, you may think we have got enough code to run a window, but if you run the program you'll see nothing will show up. That's because we're missing one more thing.
+
+The last thing we need to do is to tell the window to run. At the bottom of your `Main` method, add the following:
+
+```cs
+_window.Run();
+```
+
+And that's it! Run the program and you should hopefully see a window.
+
+![Window](../../images/opengl/chapter1/window1.png)
+
+## Window Events
+The window we've just created has several events we can subscribe to, such as loading and rendering, which are key for your program to work properly.
+
+For this tutorial, we'll be subscribing to three events:
+* Load
+* Update
+* Render
+
+Add the following three methods to your class:
+
+```cs
+private static void OnLoad() { }
+
+private static void OnUpdate(double dt) { }
+
+private static void OnRender(double dt) { }
+```
+
+These are the methods that will power our main application. For this tutorial, we'll only be using `OnLoad`, but it's helpful to have all three.
+
+> [!Note]
+> The `dt` parameter (which you will often see labelled as `obj`), is the **delta time**, which is the amount of time, in seconds, that has passed since the last frame. Using delta time is key for framerate-independent actions, such as movement.
+
+Next, we need to subscribe to these events. Add the following code to your `Main` method, just after you call `Window.Create()`:
+
+```cs
+_window.Load += OnLoad;
+_window.Update += OnUpdate;
+_window.Render += OnRender;
+```
+
+Launch your program again and you will see that... nothing has changed. Good! That means it's working correctly. Try adding some logs in the load, update, and render methods to see exactly when they are called.
+
+![Loading, Updating & Rendering](../../images/opengl/chapter1/loading-rendering.png)
+
+We're now ready to handle some input!
+
+## Handling Input
+Input is a key part of any program. If the user can't input anything, it's not a very useful program. For this tutorial, we're simply going to close the program when the user presses the escape key.
+
+### Creating an input context
+An input context is responsible for receiving user input, and handling input events. Creating one is simple, thanks to an extension method that `Silk.NET.Input` provides to the `IWindow`.
+
+In your `Load` method, add the following:
+
+```cs
+IInputContext input = _window.CreateInput();
+```
+
+This will create our input context. To handle keyboard input, we need to subscribe to the `KeyDown` event.
+
+First, let's create the method that handles this event:
+
+```cs
+private static void KeyDown(IKeyboard keyboard, Key key, int keyCode) { }
+```
+
+Now that we've done that, we can subscribe to it in our `Load` method. Add the following after you create the input context:
+
+```cs
+for (int i = 0; i < input.Keyboards.Count; i++)
+    input.Keyboards[i].KeyDown += KeyDown;
+```
```

Comment:
Hello! Consider this:
```
foreach (var keyboard in input.Keyboards)
            keyboard.KeyDown += KeyDown;
```

Reply:
I will stick with the original as the code samples uses it, this can be revisited later.